index.js

```diff
@@ -17,23 +21,72 @@ var resolveValue = function(dna, query) {
   }
 }
 
+var resolveSelfReferencePlaceholders = function (dna, valueWithPlaceholdes) {
+  var matches = valueWithPlaceholdes.match(re.selfReference).sort().filter(function (value, index, array) {
```

Comment:
@outbounder just a small tweak, the filter callback function - we can re-use it down in the referencePlaceholder switch case :)